src/Features/Core/Portable/UseNullPropagation/AbstractUseNullPropagationDiagnosticAnalyzer.cs

```diff
@@ -49,18 +50,27 @@ protected AbstractUseNullPropagationDiagnosticAnalyzer()
         protected abstract ISyntaxFactsService GetSyntaxFactsService();
         protected abstract ISemanticFactsService GetSemanticFactsService();
 
+        protected abstract bool TryAnalyzePatternCondition(SyntaxNode conditionNode, out SyntaxNode conditionLeft, out SyntaxNode conditionRight, out bool isEquals);
+
         protected override void InitializeWorker(AnalysisContext context)
         {
             context.RegisterCompilationStartAction(startContext =>
             {
                 var expressionTypeOpt = startContext.Compilation.GetTypeByMetadataName("System.Linq.Expressions.Expression`1");
+
+                var objectType = startContext.Compilation.GetSpecialType(SpecialType.System_Object);
+                var referenceEqualsMethod = objectType?.GetMembers(nameof(ReferenceEquals))
+                                      .OfType<IMethodSymbol>()
+                                      .FirstOrDefault(m => m.DeclaredAccessibility == Accessibility.Public &&
+                                                           m.Parameters.Length == 2);
+
                 startContext.RegisterSyntaxNodeAction(
-                    c => AnalyzeSyntax(c, expressionTypeOpt), GetSyntaxKindToAnalyze());
+                    c => AnalyzeSyntax(c, expressionTypeOpt, referenceEqualsMethod), GetSyntaxKindToAnalyze());
             });
 
         }
 
-        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, INamedTypeSymbol expressionTypeOpt)
+        private void AnalyzeSyntax(SyntaxNodeAnalysisContext context, INamedTypeSymbol expressionTypeOpt, IMethodSymbol referenceEqualsMethod)
```

Comment:
:bulb: This should be `referenceEqualsMethodOpt`. The `Opt` suffix indicates locally that the argument could be null. The local variables that get passed for this argument should also get renamed.

Reply:
I changed this too. Let me know if I misinterpreted anything here.